src/runtime/audience-action-local-flow.ts

```diff
@@ -304,35 +326,61 @@ export class AudienceActionLocalFlow implements AudienceActionFlow {
     }
   }
 
-  private async initRewardedAdWall_() {
-    this.eventManager_.logSwgEvent(AnalyticsEvent.IMPRESSION_REWARDED_AD);
+  private async googletagReady_(): Promise<boolean> {
+    for (let i = 0; i < this.detectGptRetries_; i++) {
+      if (this.deps_.win().googletag?.apiReady !== undefined) {
+        return this.deps_.win().googletag.apiReady;
+      }
+      await new Promise((r) => setTimeout(r, this.detectGptRetriesMs_));
+    }
+    return false;
+  }
 
-    const config = await this.getConfig_();
+  private async checkGoogletagAvailable_(): Promise<boolean> {
+    const googletagReady = await this.googletagReady_();
+    return googletagReady && !!this.deps_.win().googletag?.getVersion();
+  }
 
+  private async initRewardedAdWall_() {
+    this.eventManager_.logSwgEvent(AnalyticsEvent.IMPRESSION_REWARDED_AD);
```

Comment:
Should this be logged on successful impression?

Reply:
This should be logged for everything. In retrospect I realize that's different from how other prompts do it, but I prefer this way because it allows us to easily see everything as a portion of overall impressions.

Reply:
I'm afraid that could lead to misunderstanding in analytics. Impression by definition implies successful impressions and making an exception would create confusions for other people. I would recommend creating a separate event type to count the overall number of flow initiation, rather than using the impression event for this purpose. Also, refer to [go/swg-logged-event-types](https://goto.google.com/swg-logged-event-types)

Reply:
Ack. Moved impressions to when the prompt is ready to be shown. Don't think we need the new event yet as we can infer it from the total of the impressions and errors, but added a todo because it will be nice to have.